### PR TITLE
refs #12295 - uses USR1 signal to handle log rotation

### PR DESCRIPTION
**Pull request by witlessbird**

### CODE REVIEW

**Comment by witlessbird** [comment]
Pls. see https://github.com/theforeman/smart-proxy/pull/394 for corresponding changes in smart-proxy

**Comment by mmoll** [comment]
will there be also a PR for DEB? :heart:

**Comment by witlessbird** [comment]
I'll take a look.

**Comment by witlessbird** [comment]
pls. see https://github.com/theforeman/foreman-packaging/pull/1060 for debian-specific changes

File: foreman-proxy/foreman-proxy.init [hunk 1]
```diff
@@ -58,6 +58,20 @@ stop() {
     return $RETVAL
 }
 
+logrotate() {
+    echo -n $"Rotating logs for $prog: "
+    if [ -f ${FOREMAN_PROXY_PID} ]; then
+        killproc -p ${FOREMAN_PROXY_PID} -USR1
```
Comment (domcleal) [inline foreman-proxy/foreman-proxy.init]: This is missing an argument between the pid file and `-USR1`, as currently it's issuing a SIGTERM every time.  Adding -x shows `+ base=-USR1`, but if you instead run:

    killproc -p /var/run/foreman-proxy/foreman-proxy.pid foreman-proxy -USR1

then it logs:

    + killlevel=-USR1
    + base=foreman-proxy

and the process gets rotated.

Comment (domcleal) [inline foreman-proxy/foreman-proxy.init]: It should be `$prog` rather than plain `foreman-proxy `too.

File: foreman-proxy/foreman-proxy.init [hunk 2]
```diff
@@ -58,6 +58,20 @@ stop() {
...
+        RETVAL=$?
+    else
```
Comment (domcleal) [inline foreman-proxy/foreman-proxy.init]: I think there should be an extra `echo` after `RETVAL=$?` to ensure the output of the script is formatted correctly.

**Comment by witlessbird** [comment]
How does this look?

File: foreman-proxy/foreman-proxy.init [hunk 3]
```diff
@@ -58,6 +58,21 @@ stop() {
     return $RETVAL
 }
 
+logrotate() {
+    echo -n $"Rotating logs for $prog: "
+    if [ -f ${FOREMAN_PROXY_PID} ]; then
+        killproc -p ${FOREMAN_PROXY_PID} $prog -USR1
+        RETVAL=$?
+        echo -n
```
Comment (domcleal) [inline foreman-proxy/foreman-proxy.init]: plain `echo` here to get a newline, as `echo -n` won't print anything

**Comment by witlessbird** [comment]
Removed "-n" from ```echo```.

**Comment by domcleal** [comment]
Thanks @witlessbird!

**Pull request merged**